include/boost/polygon/detail/voronoi_predicates.hpp

```diff
@@ -1044,7 +1044,7 @@ class voronoi_predicates {
       }
     }
 
-    void pps(const site_type& site1,
+    bool pps(const site_type& site1,
```

Comment:
Just a suggestion: instead of having `pps()` (and the other ones) return a bool, could we perhaps check the circle downstream, say where `lies_outside_vertical_segment()` is called?
Something like: 
```
            if (!site_touches_circle(circle, site1) || !site_touches_circle(circle, site2) ||
                !site_touches_circle(circle, site3)) {
                return false;
            }
```
with:
```
        bool site_touches_circle(const circle_type &c, const site_type &s) {
            fpt_type v0c[2] = {c.x() - to_fpt(s.point0().x()), c.y() - to_fpt(s.point0().y())};
            if (!s.is_segment()) {
                fpt_type square_dist = v0c[0] * v0c[0] + v0c[1] * v0c[1];
                return ulp_cmp(square_dist, (c.lower_x() - c.x()) * (c.lower_x() - c.x()), ULPS) == ulp_cmp_type::EQUAL;
            }
            fpt_type v01[2] = {to_fpt(s.point1().x()) - to_fpt(s.point0().x()),
                               to_fpt(s.point1().y()) - to_fpt(s.point0().y())};
            fpt_type dot = (v0c[0] * v01[0] + v0c[1] * v01[1]) / (v01[0] * v01[0] + v01[1] * v01[1]);
            return ulp_cmp(dot, 0.0, ULPS) != ulp_cmp_type::LESS && ulp_cmp(dot, 1.0, ULPS) != ulp_cmp_type::MORE;
        }
```

However, it is probably desirable that if `circle_existence_predicate_.pps()` returns `true` then `circle_formation_functor_.pps()` has to compute a valid circle event (i.e. not a NaN and all three sites touching the circle) and we should then assert it instead of returning `false`.

Reply:
Sure, that is an option. I just added the test in `pps()` itself so I did not have to repeat the same code 3 times (in code text). 

> and we should then assert it instead of returning `false`.

Yes, this PR does not solve the root issue. It just fixes a few edge cases.